### PR TITLE
Add Github issue forms and densify some templates

### LINKED ISSUE #4735: Add Github issue forms
### Is your feature request related to a problem? Please describe

The current issue template for bug reports can sometimes be quite confusing. Especially since it's not always clear where a comment starts/ends and what someone needs to fill out.

### Describe the solution you'd like

Start using **Github issue forms** to make filing a bug report easier.

### Additional context

https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

This start using the form template that are still in beta. Result can be seen and tested here: https://github.com/Pierre-Sassoulas/pylint/issues/new/choose

Closes #4735

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    attributes:
+      value: |
+        Please make sure that this issue wasn't already requested, or already implemented in the main branch.
+  - type: textarea
+    id: what-happened
+    attributes:
+      label: What is the bug about ?
+      description:
+        Please provide the code you linted, and configurations files and multiple named
+        file if required
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:15]: We should make it as simple as possible -> shouldn't ask three things with one question. It's probably enough to ask for code that is causing the issue here.

As most errors are easily reproduceable with the default config, we should be fine adding an optional field later one for it.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      value: |
+        Given a file `a.py`:
+
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:18]: These lines are not really necessary

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        <a> = b + 1
+        ```
+
+        Using the following configuration pylintrc:
+        ```ini
+        # Please remove configuration unrelated to the bug
+
+        [MESSAGES CONTROL]
+        enable=use-symbolic-message-instead,
+        ```
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:31]: See comment above regarding the pylintrc config

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: current-behavior
+    attributes:
+      label: What is the command used and its actual output ?
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:37]: Same comment regarding the label

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      description: Please provide the command used and the current output
+      value: |
+        Result of `pylint a.py`:
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:40]: This line is also not really necessary

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    id: os
+    attributes:
+      label: Operating system ?
+      options:
+        - Linux
+        - Mac
+        - PC
+        - Other
+    validations:
+      required: true
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/BUG-REPORT.yml:76]: Should the operating system really be required? Only a handful of issue a system depended and then it's likely already part of the `bug description` above. Maybe we should even remove it completely.

File: .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+labels: [enhancement, needs triage]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Please make sure that this feature wasn't already requested or already implemented in the main branch.
+  - type: textarea
+    id: current-problem
+    attributes:
+      label: Please describe the problem this feature would solve
```
Comment (cdce8p) [inline .github/ISSUE_TEMPLATE/FEATURE-REQUEST.yml:12]: Same comment about the labels applies here as well

**Review by cdce8p — COMMENTED** [review]
Left some comments.

If you like, take a look at the `BUG` template for `setuptools`: https://github.com/pypa/setuptools/issues/new?assignees=&labels=bug%2CNeeds+Triage&template=bug-report.yml&title=%5BBUG%5D+

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml (hunk 2 above)
Comment (Pierre-Sassoulas) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml:20]: Well I think it nudges the reporter to provide a reproducible example and I really like having a reproducible example, I think I'll keep that.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml (hunk 5 above)
Comment (Pierre-Sassoulas) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml:51]: Very often the reporter do not gives the pylint output and just gives a general description "I don't like the output", I'll also keep this, that as it nudges reporter to make a report that permits to understand the problem without thinking too much or linting the code ourselves.

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml (hunk 3 above)
Comment (Pierre-Sassoulas) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml]: 👍

**Comment by Pierre-Sassoulas** [comment]
Thank you for the review and the original @cdce8p, I think these new forms are great !

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml (hunk 6 above)
Comment (Pierre-Sassoulas) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml]: I made this part optional and a free text (like in setuptools really)

**Pull request merged**